endpoint.go

```diff
@@ -289,10 +290,14 @@ func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	if err = sb.populateNetworkResources(ep); err != nil {
-		return err
+	sb.Lock()
```

Comment:
what am I missing ? heap update is not done in `populateNetworkResources`. it is in Join. Is this comment still valid ?